### PR TITLE
Moving slice to impl - slice (FX Converter Refactor [16/N]) - <Target: converter_reorg_elementwise>

### PR DESCRIPTION
**Pull request by apbose**
]

### CODE REVIEW

**Review by wushirong — APPROVED** [review]

File: py/torch_tensorrt/fx/converters/aten_ops_converters.py [hunk 1]
```diff
@@ -465,6 +502,27 @@ def aten_ops_sym_numel(
...
+    return slice_op(
+        network,
+        target,
+        SourceIR.ATEN,
+        name,
+        args[0],
+        args[1],
+        args[2],
+        args[3],
+        args[4],
```
Comment (gs-olive) [inline py/torch_tensorrt/fx/converters/aten_ops_converters.py:522]: This should check if the arguments list is long enough to enable the `step` option. Many models do not specify the `step` and it is omitted from the args list, causing an `IndexError`. Potentially the `or_none` function in this file could be modified slightly to help here:

```python
def arg_bounds_check(args, i, replacement=None):
    return args[i] if len(args) > i else replacement

...

# Replace step
arg_bounds_check(args, 4, replacement=1):
```

File: py/torch_tensorrt/fx/converters/aten_ops_converters.py [hunk 2]
```diff
@@ -465,6 +502,27 @@ def aten_ops_sym_numel(
...
+) -> Union[TRTTensor, Sequence[TRTTensor]]:
+    return slice_op(
+        network,
+        target,
+        SourceIR.ATEN,
+        name,
+        args[0],
+        args[1],
+        args[2],
+        args[3],
```
Comment (gs-olive) [inline py/torch_tensorrt/fx/converters/aten_ops_converters.py:521]: For `args[3]`, as in the TorchScript `slice` converter, the `aten` opset sometimes uses `9223372036854775807` to represent the entire dimension. We might use a check like `args[3] if args[3] != 9223372036854775807 else args[0].shape[args[1]]` to deal with this case.

File: py/torch_tensorrt/fx/converters/aten_ops_converters.py (hunk 1 above)
Comment (apbose) [reply py/torch_tensorrt/fx/converters/aten_ops_converters.py:522]: Ok I will implement this change. Do you have a model that can be run to come across this case?

File: py/torch_tensorrt/fx/converters/aten_ops_converters.py (hunk 2 above)
Comment (apbose) [reply py/torch_tensorrt/fx/converters/aten_ops_converters.py:521]: Looking at this.

File: py/torch_tensorrt/fx/converters/aten_ops_converters.py (hunk 1 above)
Comment (gs-olive) [reply py/torch_tensorrt/fx/converters/aten_ops_converters.py:522]: Yes, MobileNetV2, which is tested in the Dynamo E2E test cases elicits this scenario:
https://github.com/pytorch/TensorRT/blob/bd9c29ab046df4605107b8fc530f353c054a5fc3/py/torch_tensorrt/dynamo/test/test_dynamo_backend.py#L50-L78

**Comment by narendasan** [comment]
Closing in favor of https://github.com/pytorch/TensorRT/pull/2070

**Pull request closed without merging**